docs/pages/radiogroup.js

```diff
@@ -0,0 +1,695 @@
+          - When there is only one item to select or deselect. Use Checkbox instead.
+          - When a selection takes immediate effect, especially on mobile. Use [Switch](/switch) instead.
+          - When visually, it’s hard to tell that a RadioGroup turns something on or off. Use Switch instead.
+        `}
+          />
+        </MainSection.Subsection>
+      </MainSection>
+      <MainSection name="Best Practices" />
+      <MainSection name="Accessibility">
+        <MainSection.Subsection
+          title="Labels"
+          description="Each RadioButton in a RadioGroup should have a label that can be read by screen readers, and that can be clicked or tapped to make it easier for users to select and deselect options. Therefore, make sure to supply the label prop. If that’s not possible, make sure your stand-alone Label has an `htmlFor` prop that matches the `id` of the RadioButton. Test that a RadioButton and label are properly connected by clicking or tapping on the label and confirming that it activates the RadioButton next to it."
+        />
+        <MainSection.Subsection
+          title="Legends"
+          description={`Each RadioGroup should have a legend that clearly delineates what is being chosen. If you cannot use the provided legend styling, \`legendDisplay\` can be set to \`hidden\`, and an alternative legend can be displayed.`}
+        />
+        <MainSection.Subsection
+          title="Keyboard interaction"
+          description={`After focus has been set on the first RadioButton inside a RadioGroup, the arrow keys are used to cycle focus between the various options. Clicking or tapping the label of RadioButton should also focus that particular RadioButton. `}
+        />
+        <MainSection.Subsection
+          title="Error message"
+          description={`If RadioGroup has an \`errorMessage\`, the \`id\` on RadioGroup must also be set to properly link the error message to RadioGroup.
+    `}
+        />
+      </MainSection>
+      <MainSection
+        name="Localization"
+        description={`Be sure to localize \`errorMessage\`, \`subtext\`, \`label\`, and \`legend\`. Be mindful of label length so that it doesn’t truncate in languages with lengthier character counts.`}
+      />
+
+      <MainSection name="Variants">
+        <MainSection.Subsection
+          title="Direction"
+          description="RadioGroups can be shown in a column or row by specifying the `direction` property."
+        >
+          <MainSection.Card
+            defaultCode={`
+function RadioButtonExample() {
+  const [favorite, setFavorite] = React.useState(undefined);
+  const [favoriteFood, setFavoriteFood] = React.useState(undefined);
+
+  return (
+    <Flex gap={8}>
+      <RadioGroup legend="What is your favorite pet?">
+        <RadioGroup.RadioButton
+          checked={favorite === 'dogs'}
+          id="favoriteDog"
+          label="Dogs"
+          name="favorite"
+          onChange={() => setFavorite( 'dogs' )}
```

Comment:
nit: spaces around the arg don't match our prettier config (here and throughout)